ncmcm/bundlenet/bundlenet.py

```diff
@@ -168,37 +192,44 @@ def train_model(x_train, b_train_1, model, b_type, gamma, learning_rate, n_epoch
         n_epochs (int): Number of training epochs.
         initialisation (str): 'pca_init' or 'best_of_5_init'
         validation_data: (x_test, b_test_1)
+        device (torch.device): Device where the model should be trained.
     Returns:
         numpy.ndarray: Array of loss values during training.
     """
-    train_dataset = tf_batch_prep(x_train, b_train_1)
-    optimizer = tf.keras.optimizers.Adam(learning_rate=learning_rate)
+    if device is None:
+        device = torch.device('cpu')
```

Comment:
Maybe it's best to check if the machine has a gpu and use it as the default, if not, use the cpu? something along the lines of `device = torch.device("cuda" if torch.cuda.is_available() else "cpu")`

Reply:
Yeah for sure, I updated it.
Then in this case, I also had to add ``device = next(model.parameters()).device`` to the demos, because for the evaluation the tensor should be created wherever the model is